cob_obstacle_distance/include/cob_obstacle_distance/marker_shapes_impl.hpp

```diff
@@ -0,0 +1,142 @@
+/*
+ *****************************************************************
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2015 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_obstacle_distance
+ *
+ * \author
+ *   Author: Marco Bezzon, email: Marco.Bezzon@ipa.fraunhofer.de
+ *
+ * \date Date of creation: May, 2015
+ *
+ * \brief
+ *   This header contains the declaration and template implementation of MarkerShapes
+ *   which represent a combination of ROS markers and FCL geometric shapes.
+ *
+ ****************************************************************/
+
+#ifndef MARKER_SHAPES_IMPL_HPP_
+#define MARKER_SHAPES_IMPL_HPP_
+
+#include "cob_obstacle_distance/marker_shapes.hpp"
+
+/* BEGIN MarkerShape ********************************************************************************************/
+template <typename T>
+MarkerShape<T>::MarkerShape(T &fcl_object,
+      double x, double y, double z,
+      double quat_x, double quat_y, double quat_z, double quat_w,
+      double color_r, double color_g, double color_b, double color_a) : fcl_marker_converter_(fcl_object), is_drawn_(false)
```

Comment:
can we not combine (x,y,z) into a pos-type, (quat_x,quat_y,quat_z,quat_w) into a quat-type and (color_r,color_g,color_b,color_a) into an rgba-type instead of having 11 doubles?...as your providing several conversions anyway...